tests/unit/test_timeout.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2017-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""
+This class tests the timeout.py class in the integration tests.
+This is to prevent regressions that cause the timeout function to hide failed tests.
+"""
+from __future__ import absolute_import
+
+import pytest
+from mock import MagicMock, patch
+
+from tests.integ.timeout import (
+    timeout,
+    timeout_and_delete_endpoint_by_name,
+    timeout_and_delete_model_with_transformer,
+)
+
+
+EXCEPTION_MESSAGE = "This Exception is expected and should not be swallowed by the timeout."
+
+
+def test_timeout_fails_correctly_when_calling_test_throws_exception():
```

Comment:
let's add a success case and a timeout case.

Reply:
Good catch!
I added those, as well as a test for the resource deletion, as it turns out that was executing retries despite initial successes, adding 30s per integration test (IT) calling the timeout function (63 instances * 30 seconds ~= 30 minutes). Since the ITs run in parallel, we won't actually see the ITs run 30min faster, though.